discovery_engine_core/core/src/mab.rs

```diff
@@ -55,6 +55,9 @@ pub(crate) trait Bucket<T> {
 
     /// Removes the next best element from this bucket and returns it, or `None` if it is empty.
     fn pop(&mut self) -> Option<T>;
+
```

Comment:
Why we need to modify the bucket? This method has nothing to do with it.

Reply:
it's just an additional impl which could be useful, but we can impl it later on if we need it, i'll remove it.